### DISCUSSION TITLE
Does Tesla need permanent access to Home Assistant from the Internet?

### DISCUSSION THREAD

**Discussion by goodyear77** [llamafilm/tesla-http-proxy-addon#75] [Q&A] [answered]
Hi, I read the wiki and have successfully setup the Tesla HTTP Proxy using a cloudflare tunnel.

But I’m uneasy allowing access to my Home Assistant machine directly from the Internet, so my question is if the access needs to be permanent or only during enrollment?

There is no need for Tesla to access HA, it’s only HA that polls the Tesla API, so I’m thinking inbound Internet access to Home Assistant can be closed after the initial setup?

**Comment by baylanger** [comment, marked as answer]
Please take the time to read the DOCS in case you missed something. See [prerequisites](https://github.com/llamafilm/tesla-http-proxy-addon/blob/main/tesla_http_proxy/DOCS.md#prerequisites)

**Reply by goodyear77** [reply]
Thanks! All clear 😊

**Reply by baylanger** [reply]
FYI just created this [PR](https://github.com/llamafilm/tesla-http-proxy-addon/pull/76) to make this more obvious.
